src/components/obb-clipping.js

```diff
@@ -0,0 +1,402 @@
+
+      trackedObject3D.parent.matrixWorld.decompose(
+        auxPosition,
+        auxQuaternion,
+        auxScale
+      );
+      auxMatrix.compose(auxPosition, identityQuaternion, auxScale);
+      trackedObject3D.parent.matrixWorld.copy(auxMatrix);
+
+      // Calculate bounding box size.
+      boundingBox.setFromObject(trackedObject3D, true);
+      boundingBox.getSize(boundingBoxSize);
+
+      // Enforce minimum dimensions.
+      boundingBoxSize.x =
+        boundingBoxSize.x < minimumColliderDimension
+          ? minimumColliderDimension
+          : boundingBoxSize.x;
+      boundingBoxSize.y =
+        boundingBoxSize.y < minimumColliderDimension
+          ? minimumColliderDimension
+          : boundingBoxSize.y;
+      boundingBoxSize.z =
+        boundingBoxSize.z < minimumColliderDimension
+          ? minimumColliderDimension
+          : boundingBoxSize.z;
+
+      // Restore rotations.
+      trackedObject3D.parent.matrixWorld.compose(
+        auxPosition,
+        auxQuaternion,
+        auxScale
+      );
+      this.trackedObject3D.rotation.copy(auxEuler);
+    };
+  })(),
+
+  checkTrackedObject: function () {
+    var trackedElement = document.querySelector(
+      this.data.clippingSourceSelectorString
+    );
+    if (trackedElement) {
+      this.trackedObject3D = trackedElement.object3D;
+      this.updateCollider();
+    }
+    // console.log('trackedElement', this.trackedObject3D);
+    return this.trackedObject3D;
+  },
+
+  tick: (function () {
+    var auxPosition = new THREE.Vector3();
+    var auxScale = new THREE.Vector3();
+    var auxQuaternion = new THREE.Quaternion();
+    var auxMatrix = new THREE.Matrix4();
+
+    return function () {
+      var obb = this.obb;
+      var renderColliderMesh = this.renderColliderMesh;
+      var trackedObject3D = this.checkTrackedObject();
+
+      if (!trackedObject3D) {
+        return;
+      }
+
+      trackedObject3D.updateMatrix();
+      trackedObject3D.updateMatrixWorld(true);
+      trackedObject3D.matrixWorld.decompose(
+        auxPosition,
+        auxQuaternion,
+        auxScale
+      );
+
+      // Recalculate collider if scale has changed.
+      if (
+        Math.abs(auxScale.x - this.previousScale.x) > 0.0001 ||
+        Math.abs(auxScale.y - this.previousScale.y) > 0.0001 ||
+        Math.abs(auxScale.z - this.previousScale.z) > 0.0001
+      ) {
+        this.updateCollider();
+      }
+
+      this.previousScale.copy(auxScale);
+
+      // reset scale, keep position and rotation
+      auxScale.set(1, 1, 1);
+      auxMatrix.compose(auxPosition, auxQuaternion, auxScale);
+      // Update OBB visual representation.
+      if (renderColliderMesh) {
+        renderColliderMesh.matrixWorld.copy(auxMatrix);
+      }
+
+      // Reset OBB with AABB and apply entity matrix. applyMatrix4 changes OBB internal state.
+      obb.copy(this.aabb);
+      obb.applyMatrix4(auxMatrix);
+
+      const clipPlanes = this.createPlanesFromOBB(obb);
```

Comment:
you don't need to update the clipping planes every frame. Can do similar to bounding box recalculation. Consider scenarios when you want to recalculate: rotation, scale, position changes

Reply:
```
    const clipPlanes = this.createPlanesFromOBB(obb);
    this.applyClippingPlanes(clipPlanes);
```